### DISCUSSION TITLE
Adding a new molecular fingerprint to RDKit?

### DISCUSSION THREAD

**Discussion by hilsan** [rdkit/rdkit#7534] [Ideas]
As a part of a research project on machine learning for molecular atmospheric chemistry, my research group has developed and tested a customized version of the MACCS fingerprint, tailored to atmospheric compounds and their property prediction. 

We are currently working to publish this descriptor so that it is available for the broader atmospheric chemistry community.  We are wondering if it is possible to add our descriptor to RDKit and join/or collaborate with the developers? If not, do you have any recommendations or examples of these types of descriptors which make use of RDKit but are published elsewhere? 

 Our new descriptor adds approx 100 new features to MACCS  (making the total number of features upwards of 270). The new features are all either 1) based on SMARTS pattern matching for fragments typical in atmospheric compounds (eg. peroxide or nitrate groups) or 2) counts of certain elements or ring fragments. 
 
 Let us know your thoughts on this idea and how you recommend that we should proceed! 
 
 Many thanks, Hilda

**Comment by kuulia** [comment]
[ATMOMACCS.zip](https://github.com/user-attachments/files/16026743/ATMOMACCS.zip)

As part of this idea we've worked on a Python script that is based on the blueprint of MACCSKeys.py. We have got a working demo implementation here, where dependencies are restricted to RDKit and numpy. There are a few edge cases still where the substructure matching is behaving slightly differently to our original implementation, which included other libraries and tools. 

Similarly to MACCS, this could also be implemented in C++ in a very parallel way, to which we would hope to receive some help from the other developers if there is interest in incorporating this descriptor to RDKit. 

Thanks, Linus.
